File: src/main/webapp/src/app/models/action-trigger-midi-note-on.ts
import { ActionTriggerMidi } from "./action-trigger-midi";

export class ActionTriggerMidiNoteOn extends ActionTriggerMidi {
  note: number;

  constructor(data?: any) {
    super(data);

    if (!data) {
      return;
    }

    this.note = data.note;

    if(!this.note) {
      this.note = 0;
    }
  }

  toJSON() {
    return { actionTriggerMidiNoteOn: { ...this } };
  }
}

File: src/main/java/com/ascargon/rocketshow/midi/ActionTriggerMidiNoteOn.java
package com.ascargon.rocketshow.midi;

import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class ActionTriggerMidiNoteOn extends ActionTriggerMidi {

    // The note, this action should be triggered. If null -> all notes
    private Integer note;

}
